modules/masterbar/masterbar.php

```diff
@@ -31,6 +31,11 @@ function __construct() {
 			return;
 		}
 
+		// Don't show the masterbar on WordPress mobile apps.
+		if ( preg_match( '/wp-(android|iphone)/', $_SERVER['HTTP_USER_AGENT'] ) ) {
```

Comment:
does it worth making this checking stronger? I did a quick research and maybe [it's useful](https://stackoverflow.com/questions/5335237/how-to-check-if-the-request-came-from-mobile-or-computer-in-php).

Reply:
Also [wp_is_mobile()](https://core.trac.wordpress.org/browser/tags/5.1.1/src/wp-includes/vars.php#L136) is even better :-D

Reply:
We're checking specifically for webviews from the WordPress iOS or Android apps, not any mobile browser

Reply:
> The aim of this change is to prevent users from navigating to pages they can access through the app.

So it's just needed for iOS and Android? does it not happen for other mobile OS?

Reply:
It's fine if they're shown in mobile browsers. This is only meant to be hidden when accessed from within the WordPress iOS or Android apps.

Reply:
fwiw we do have helper functions in Jetpack that can be used if we wanted to improve that check:
https://github.com/Automattic/jetpack/blob/f098dbe15b348e8d41a1ef97ea65b4e8d7224900/class.jetpack-user-agent.php

While that may not be needed here, I would recommend checking for `! empty( $_SERVER['HTTP_USER_AGENT'] )` before to actually use it.

Reply:
`Jetpack_User_Agent_Info:: is_mobile_app()` seems to be the appropriate one

Reply:
Didn't know about `Jetpack_User_Agent_Info::is_mobile_app()`. Thanks for the suggestion! Refactored the change to use it in fd1c2cf.